common/encryption/encrypt.h

```diff
@@ -52,7 +52,10 @@ std::vector<unsigned char*> encrypt_bytes(std::string model_data) {
         tag
     );
 
-    std::vector<unsigned char*> out_iv_tag{output, iv, tag};
+    unsigned char** out_iv_tag = (unsigned char**) malloc(3 * sizeof(unsigned char*));
+    *out_iv_tag = output;
+    *(out_iv_tag + 1) = iv;
+    *(out_iv_tag + 2) = tag;
 
     return out_iv_tag;
 }

```

Comment:
Don't think you actually wrote this code, but can you please also change `decrypt_bytes` to take in an array that will hold the plaintext once decryption occurs? We can also do this in a later PR if that's easier, up to you.

Reply:
I'll do this now -- I think this function actually has a bug anyway because `out` is being stored on the stack (see comment above)